text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,434 @@
+# RFC - Glue CDK L2 Construct
+
+## L2 Construct for AWS Glue Connections, Jobs, and Workflows
+
+* **Original Author(s):** @natalie-white-aws, @mjanardhan @parag-shah-aws
+* **Tracking Issue:**
+* **API Bar Raiser:** @TheRealAmazonKendra
+[Link to RFC Issue](https://github.com/aws/aws-cdk-rfcs/issues/497)
+
+## Working Backwards - README
+
+[AWS Glue](https://aws.amazon.com/glue/) is a serverless data integration
+service that makes it easier to discover, prepare, move, and integrate data
+from multiple sources for analytics, machine learning (ML), and application
+development. Glue was released on 2017/08.
+[Launch](https://aws.amazon.com/blogs/aws/launch-aws-glue-now-generally-available/)
+
+Today, customers define Glue data sources, connections, jobs, and workflows
+to define their data and ETL solutions via the AWS console, the AWS CLI, and
+Infrastructure as Code tools like CloudFormation and the CDK. However, they
+have challenges defining the required and optional parameters depending on
+job type, networking constraints for data source connections, secrets for
+JDBC connections, and least-privilege IAM Roles and Policies. We will build
+convenience methods working backwards from common use cases and default to
+recommended best practices.
+
+This RFC proposes updates to the L2 construct for Glue which will provide
+convenience features and abstractions for the existing
+[L1 (CloudFormation) Constructs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/AWS_Glue.html) building on the
+functionality already supported in the [@aws-cdk/aws-glue-alpha module](https://github.com/aws/aws-cdk/blob/v2.51.1/packages/%40aws-cdk/aws-glue/README.md).
+
+## Create a Glue Job
+
+The glue-alpha-module already supports three of the four common types of Glue
+Jobs: Spark (ETL and Streaming), Python Shell, Ray. This RFC will add the
+more recent Flex Job. The construct also implements AWS practice
+recommendations when creating a Glue Job such use of Secrets Management for
+Connection JDBC strings, Glue Job Autoscaling, least privileges in terms of
+IAM permissions and also sane defaults for Glue job specification (more details
+are mentioned in the below table).
+
+This RFC will introduce breaking changes to the existing glue-alpha-module to
```

Comment:
Can you call out these breaking changes when you introduce them in the text?

Reply:
The breaking changes will be related to renaming functions to be more accurately linked to the job types and languages supported, function parameters, and new restrictions on Python and Glue versions (not supporting deprecated versions).  We could add a sentence for each job type saying this maps to [existing alpha function] if that would help clarify the specific breaking changes, but Kendra said the expectation was that since it's an alpha, there will potentially be breaking changes until it's stabilized, and didn't ask for a comprehensive migration guide.